apps/emqx/rebar.config

```diff
@@ -66,7 +66,7 @@
     {plt_location, "."},
     {plt_prefix, "emqx_dialyzer"},
     {plt_apps, all_apps},
-    {plt_extra_apps, [hocon]},
+    {plt_extra_apps, [hocon,mnesia,bcrypt,os_mon,pbkdf2,emqx_http_lib, recon]},
```

Comment:
Strange that we now need to do that, some of them probably need to be added to the `emqx.app.src`?

Reply:
mainly for two reasons
1. apps (e.g. mnesia) need to be added to all apps which calls mnesia, this is a bit annoying
2. apps (e.g. hocon, bcrpyt(when using nif), emqx_http_lib) they are static libs dummy root sup or no root sup at all.

Reply:
Understood.

>  they are static libs dummy root sup or no root sup at all.

Even then, a lot of stuff treats entries in app.src as _dependencies that need to be taken into account_, not just dialyzer (i.e., relx).

Reply:
Sorry. this is actually just for the standalone app.
Will fix it in the next PR.

Reply:
and maybe bcrypt is not needed to be dynamic dep since we dropped windows.

Reply:
https://github.com/emqx/emqx/pull/12179